sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -105,6 +105,12 @@ abstract class Expression extends TreeNode[Expression] {
       val isNull = ctx.freshName("isNull")
       val value = ctx.freshName("value")
       val eval = doGenCode(ctx, ExprCode("", isNull, value))
+      eval.isNull = if (this.nullable) eval.isNull else "false"
```

Comment:
any findings for the more aggressive approach? `val isNull = if (this.nullable) ...`

Reply:
Compilation error like:
```
[info]   Cause: org.codehaus.commons.compiler.CompileException: File 'generated.java', Line 59, Column 14: failed to compile: org.codehaus.commons.compiler.CompileException: File 'generated.java', Line 59, Column 14: Identifier expected instead of 'false'
```

Seems some expressions uses `eval.isNull` as lvalue. Do you think we should find them out and change them?

Reply:
at least we should do it in a new PR. do you wanna do it after this one get merge?

Reply:
Yea, I plan to do that after this.

Reply:
BTW after some glance, this may not worth. It's annoying to check `nullable` and deal with how to assign `isNull` in `Expressin.doGenCode`

Reply:
So, shall we keep it as it is or restore it back?

Reply:
oh I mean the more aggressive approach may not worth, we can even simplify some `Expression.doGenCode` that try to set `isNull` to `false`.